ddsp/ai-platform/README.md

```diff
@@ -0,0 +1,103 @@
+* `--master-image-uri` - URI of the Docker image you've built and submitted to Container Registry
+
+##### Program flags:
+* `--save_dir` - **Mandatory flag**. Bucket directory where checkpoints and summary events will be saved during training.
+* `--file_pattern` - **Mandatory flag**. Pattern of the data files names. Must include a whole bucket directory.
+* `--restore_dir` - Bucket directory from which checkpoints will be restored before training. When not provided defaults to save_dir. If there are no checkpoints in the given directory, training will resume.
+* `--batch_size` - Batch size.
+* `--learning_rate` - Learning rate.
+* `--num_steps` - Number of training steps.
+* `--early_loss_stop_value` - Early stopping. When the total_loss reaches below this value training stops.
+* `--steps_per_save` - Steps per model save.
+* `--restore_per_summary` - Steps per training summary save.
+* `--hypertune` - If True enables metric reporting for hyperparameter tuning.
+
+##### Additional configuration flags
+* `--gin_param` - Gin parameter bindings. Using this flag requires some familiarity with the Magenta DDSP source code. Take a look at parameters you can specify in [Gin config files](https://github.com/magenta/ddsp/tree/master/ddsp/training/gin).
+* `--gin_search_path` - Additional gin file search path. Must be path inside Docker container and necessary gin configs should be added at the Docker image building stage.
+* `--gin_file` - Additional Gin config file. If the file is in gstorage bucket specify a whole gstorage path.
+
+
+You can add your own Gin config files in two ways:
+* Add Gin config file to the gstorage bucket and specify `--gin-file` as a gstorage path:
+ ```bash 
+ --gin_file=gs://[YOUR_STORAGE_BUCKET_NAME]/[PATH_IN_STORAGE_BUCKET]/config_file.gin
+ ```
+ * Add copying local Gin configs to the Dockerfile, build and push the image. Specify `--gin_search_path` flag as a directory inside the Docker container where gin file is located and `--gin-file` as a copied file name.
+
+
+### Note on cluster configuration and hyperparameters
+
+There are two cluster configurations prepared:
+* `config_single_vm.yaml` - 1 VM configuration with 1 NVIDIA Tesla T4 GPUs. Training with this configuration and recommended parameters (*batch_size: 16, learning_rate:0.0001, num_steps:40000, early_stop_loss_value:5.0*) takes around 10 hours and consumes around 19 [ML units](https://cloud.google.com/ai-platform/training/pricing#ml-units).
+
+* `config_multiple_vms.yaml` - 4 VM configuration with 8 NVIDIA Tesla T4 GPUs. Training with this configuration and recommended parameters (*batch_size: 128, learning_rate:0.001, num_steps:15000, early_stop_loss_value:5.0*) takes around 5 hours and consumes around 44 [ML units](https://cloud.google.com/ai-platform/training/pricing#ml-units).
+
+Feel free to experiment and define your [cluster configurations](https://cloud.google.com/ai-platform/training/docs/using-gpus).
+
+### Note on dataset location
```

Comment:
can you comment on why you might want to do this? doesn't the docker image ultimately get uploaded to GCS anyway?

Reply:
It's just a second option of dataset placement. The user either uploads the data to GCS or they build the image with the data inside the container. For clarity of the documentation I can remove this section. I just wanted to let the user know that this approach works as well.

Reply:
Done